krita: split unwrapped package, prepare for qt6 build

sifmelcara opened the pull request:
This addresses the complaint about not able to have krita's build environment when using nix shell, and also refactors for krita's qt6 build.

1. Unwrapped package is separated so user can access the build environment with nix shell
2. Avoids the usage of `libsForQt5.callPackage`, organize qt dependency for easier switch to qt6.

This change should be no-op, although the nix derivation hash changed, I checked the krita binary and compare with the old version, there is only 31 bytes difference (probably some kind of hash reference?). Also tested the binary to make sure it still works.

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform:
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- Tested, as applicable:
  - [ ] [NixOS tests] in [nixos/tests].
  - [ ] [Package tests] at `passthru.tests`.
  - [ ] Tests in [lib/tests] or [pkgs/test] for functions and "core" functionality.
- [x] Ran `nixpkgs-review` on this PR. See [nixpkgs-review usage].
- [x] Tested basic functionality of all binary files, usually in `./result/bin/`.
- Nixpkgs Release Notes
  - [ ] Package update: when the change is major or breaking.
- NixOS Release Notes
  - [ ] Module addition: when adding a new NixOS module.
  - [ ] Module update: when the change is significant.
- [x] Fits [CONTRIBUTING.md], [pkgs/README.md], [maintainers/README.md] and other READMEs.

[NixOS tests]: https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests
[Package tests]: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests
[nixpkgs-review usage]: https://github.com/Mic92/nixpkgs-review#usage

[CONTRIBUTING.md]: https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md
[lib/tests]: https://github.com/NixOS/nixpkgs/blob/master/lib/tests
[maintainers/README.md]: https://github.com/NixOS/nixpkgs/blob/master/maintainers/README.md
[nixos/tests]: https://github.com/NixOS/nixpkgs/blob/master/nixos/tests
[pkgs/README.md]: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md
[pkgs/test]: https://github.com/NixOS/nixpkgs/blob/master/pkgs/test

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

lelgenio reviewed (commented):
The only thing I noticed is that people who use `krita.unwrapped` will need to use `krita.krita-unwrapped`

https://github.com/search?q=lang%3Anix+krita.unwrapped&type=code

otherwise LGTM

sifmelcara commented:
> The only thing I noticed is that people who use `krita.unwrapped` will need to use `krita.krita-unwrapped`
> 
> https://github.com/search?q=lang%3Anix+krita.unwrapped&type=code
> 

Thanks! Fixed

lelgenio reviewed (approved):
LGTM 😄👍

sifmelcara committed:
krita: split unwrapped package, prepare for qt6 build

This addresses the complaint about not able to have krita's build
environment when using nix shell, and also refactors for krita's qt6
build.

nixos-discourse commented:
This pull request has been mentioned on **NixOS Discourse**. There might be relevant details there:

https://discourse.nixos.org/t/prs-ready-for-review/3032/6215

SuperSandro2000 commented on pkgs/by-name/kr/krita/default.nix:
Removing that file is neat.

SuperSandro2000 commented:
> Unwrapped package is separated so user can access the build environment with nix shell

That already worked: ``nix shell nixpkgs#krita.unwrapped``
